packages/plugin-lighthouse/src/lib/constants.ts

```diff
@@ -0,0 +1,36 @@
+import { type Config, type IcuMessage, defaultConfig } from 'lighthouse';
+import { Audit, Group } from '@code-pushup/models';
+
+export const LIGHTHOUSE_PLUGIN_SLUG = 'lighthouse';
+export const LIGHTHOUSE_REPORT_NAME = 'lighthouse-report.json';
+
+type Meta = { title: IcuMessage; description: IcuMessage };
+const { audits, categories } = defaultConfig;
+export const GROUPS: Group[] = Object.entries(
+  categories as Record<string, Config.CategoryJson>,
+).map(([id, category]) => {
+  const meta = category as Meta & {
+    auditRefs: Config.AuditRefJson[];
+  };
+  return {
+    slug: id,
+    title: meta.title.formattedDefault,
+    refs: meta.auditRefs.map(ref => ({ slug: ref.id, weight: ref.weight })),
+  };
+});
+
+export const AUDITS: Audit[] = await Promise.all(
+  (audits as string[]).map(async path => {
+    const { default: audit } = (await import(
+      `lighthouse/core/audits/${path}.js`
+    )) as { default: unknown };
+    const { meta } = audit as { meta: Meta & { id: string } };
+    return {
+      slug: meta.id,
+      title: meta.title.formattedDefault || (meta.title as unknown as string),
+      description:
+        meta.description.formattedDefault ||
+        (meta.description as unknown as string),
+    };
+  }),
+);
```

Comment:
The type-casting is kinda wild here :grimacing: The main cause is the `defaultConfig` type definition is bit wider than the actual values seem to be. But I was able to refactor it to a more defensive approach:

![image](https://github.com/code-pushup/cli/assets/34691111/b113ff10-ac34-4278-acdd-3a5a57142d4d)


```suggestion
import {
  type Config,
  type IcuMessage,
  Audit as LHAudit,
  defaultConfig,
} from 'lighthouse';
import { Audit, Group } from '@code-pushup/models';

export const LIGHTHOUSE_PLUGIN_SLUG = 'lighthouse';
export const LIGHTHOUSE_REPORT_NAME = 'lighthouse-report.json';

const { audits, categories } = defaultConfig;

export const GROUPS: Group[] = Object.entries(categories ?? {}).map(
  ([id, category]) => ({
    slug: id,
    title: getMetaString(category.title),
    ...(category.description && {
      description: getMetaString(category.description),
    }),
    refs: category.auditRefs.map(ref => ({ slug: ref.id, weight: ref.weight })),
  }),
);

export const AUDITS: Audit[] = await Promise.all(
  (audits ?? []).map(async value => {
    const audit = await loadLighthouseAudit(value);
    return {
      slug: audit.meta.id,
      title: getMetaString(audit.meta.title),
      description: getMetaString(audit.meta.description),
    };
  }),
);

function getMetaString(value: string | IcuMessage): string {
  if (typeof value === 'string') {
    return value;
  }
  return value.formattedDefault;
}

async function loadLighthouseAudit(
  value: Config.AuditJson,
): Promise<typeof LHAudit> {
  if (typeof value === 'object' && 'implementation' in value) {
    return value.implementation;
  }
  if (typeof value === 'function') {
    return value;
  }
  const path = typeof value === 'string' ? value : value.path;
  const module = (await import(`lighthouse/core/audits/${path}.js`)) as {
    default: typeof LHAudit;
  };
  return module.default;
}
```

Reply:
Sweet! 

Yes it was wild. 😇

Thanks a lot for the refactoring!! 🎯❤️
I will merge it in and try to add some comments to the cases in `loadLighthouseAudit`.

Reply:
added comments